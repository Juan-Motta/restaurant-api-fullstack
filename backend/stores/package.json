{
  "name": "orders",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "test:coverage": "jest --coverage",
    "dev:server": "ts-node-dev --respawn --project tsconfig.json src/app.ts",
    "dev:listener": "ts-node-dev --respawn --project tsconfig.json src/listener.ts",
    "build": "tsc",
    "prod:listener": "node ./dist/src/listener.js",
    "format": "prettier --write \"src/**/*.ts\" \"tests/**/*.ts\""
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/amqplib": "^0.10.6",
    "@types/jest": "^29.5.14",
    "@types/jsonwebtoken": "^9.0.8",
    "@types/node": "^22.13.4",
    "@types/pg": "^8.11.11",
    "jest": "^29.7.0",
    "prettier": "3.5.1",
    "ts-jest": "^29.2.5",
    "ts-node-dev": "^2.0.0",
    "typescript": "^5.7.3"
  },
  "dependencies": {
    "amqplib": "^0.10.5",
    "dotenv": "^16.4.7",
    "jsonwebtoken": "^9.0.2",
    "pg": "^8.13.3"
  }
}
