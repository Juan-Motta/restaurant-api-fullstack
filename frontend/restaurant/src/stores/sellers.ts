import { ref } from 'vue'
import { defineStore } from 'pinia'
import type { SellersData } from '@/types/sellers'

export const useSellersStore = defineStore('sellers', () => {
  const sellers = ref<SellersData[]>([])

  const setSellersData = (newSellers: SellersData[]) => {
    newSellers.forEach((newSeller) => {
      const existingSellerIndex = sellers.value.findIndex((seller) => seller.id === newSeller.id)
      if (existingSellerIndex > -1) {
        sellers.value[existingSellerIndex] = { ...newSeller, ...sellers.value[existingSellerIndex] }
      } else {
        sellers.value.push(newSeller)
      }
    })
  }

  const getSortedSellersByCount = () => {
    return [...sellers.value].sort((a, b) => b.count - a.count)
  }

  const addPoint = (id: number, points: number) => {
    sellers.value = sellers.value.map((seller) => {
      if (seller.id === id) {
        return {
          ...seller,
          count: seller.count + points,
        }
      }
      return seller
    })
  }

  const resetPoints = () => {
    sellers.value = sellers.value.map((seller) => {
      return {
        ...seller,
        count: 0,
      }
    })
  }

  const getTotalPoints = () => {
    return sellers.value.reduce((total, seller) => total + seller.count, 0)
  }

  return { sellers, setSellersData, getSortedSellersByCount, addPoint, resetPoints, getTotalPoints }
})
