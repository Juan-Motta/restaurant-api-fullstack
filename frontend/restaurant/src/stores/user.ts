import { ref } from 'vue'
import { defineStore } from 'pinia'

export const useUserStore = defineStore('user', () => {
  const user = ref<{ id: number; name: string }>({ id: 0, name: '' })

  const setUserData = (data: { id: number; name: string }) => {
    user.value = data
  }

  const getUserData = () => {
    return user.value
  }

  const clearUserData = () => {
    user.value = { id: 0, name: '' }
  }

  return { user, setUserData, getUserData, clearUserData }
})
