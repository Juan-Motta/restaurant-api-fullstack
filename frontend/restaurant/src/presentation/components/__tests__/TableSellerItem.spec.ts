import { describe, it, expect, beforeEach } from 'vitest'
import { mount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import TableSellerItem from '@/presentation/components/TableSellerItem.vue'
import { useSellersStore } from '@/stores/sellers'

const $t = () => {}

describe('TableSellerItem.vue', () => {
  // Setup the Pinia store
  beforeEach(() => {
    const pinia = createPinia()
    setActivePinia(pinia)
  })

  // Test if the component renders correctly
  it('should render correctly', () => {
    const wrapper = mount(TableSellerItem, {
      global: {
        mocks: {
          $t,
        },
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the table with seller data', () => {
    const sellersStore = useSellersStore()
    sellersStore.setSellersData([
      { id: 1, name: 'Seller One', identification: 'NID1', count: 10, observations: '' },
      { id: 2, name: 'Seller Two', identification: 'NID2', count: 20, observations: '' },
    ])

    // Act: Mount the component
    const wrapper = mount(TableSellerItem, {
      global: {
        mocks: {
          $t,
        },
      },
    })

    // Assert: Check if the table renders correctly
    const rows = wrapper.findAll('tbody tr')
    // Verify that 2 rows are rendered
    expect(rows).toHaveLength(2)
    // Verify the content of the first row
    expect(rows[0].find('td:nth-child(1)').text()).toBe('2')
    expect(rows[0].find('td:nth-child(2)').text()).toBe('Seller Two')
    expect(rows[0].find('td:nth-child(3)').text()).toBe('NID2')
    expect(rows[0].find('td:nth-child(4)').text()).toBe('20')
    // Verify the content of the second row
    expect(rows[1].find('td:nth-child(1)').text()).toBe('1')
    expect(rows[1].find('td:nth-child(2)').text()).toBe('Seller One')
    expect(rows[1].find('td:nth-child(3)').text()).toBe('NID1')
    expect(rows[1].find('td:nth-child(4)').text()).toBe('10')
  })
})
