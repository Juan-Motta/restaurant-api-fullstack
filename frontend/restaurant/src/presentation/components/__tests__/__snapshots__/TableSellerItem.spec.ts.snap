// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TableSellerItem.vue > should render correctly 1`] = `
"<div data-v-acdd6653="" class="bg-white/50 py-8 px-5 border-shadow rounded-xl">
  <table data-v-acdd6653="" class="min-w-full">
    <thead data-v-acdd6653="">
      <tr data-v-acdd6653="" class="text-left">
        <th data-v-acdd6653="" class="pt-2 pb-6 px-4"></th>
        <th data-v-acdd6653="" class="pt-2 pb-6 px-4"></th>
        <th data-v-acdd6653="" class="pt-2 pb-6 px-4"></th>
        <th data-v-acdd6653="" class="pt-2 pb-6 px-4"></th>
      </tr>
    </thead>
    <tbody data-v-acdd6653=""></tbody>
  </table>
</div>"
`;
